tests/catalog/test_sql.py

```diff
@@ -1117,17 +1117,30 @@ def test_create_namespace_with_empty_identifier(catalog: SqlCatalog, empty_names
         lazy_fixture("catalog_sqlite"),
     ],
 )
-@pytest.mark.parametrize("namespace_list", [lazy_fixture("database_list"), lazy_fixture("hierarchical_namespace_list")])
-def test_list_namespaces(catalog: SqlCatalog, namespace_list: List[str]) -> None:
+def test_list_namespaces(catalog: SqlCatalog) -> None:
+    namespace_list = ["db", "db.ns1", "db.ns1.ns2", "db.ns2", "db2", "db2.ns1", "db%"]
     for namespace in namespace_list:
         catalog.create_namespace(namespace)
-    # Test global list
+
     ns_list = catalog.list_namespaces()
-    for namespace in namespace_list:
-        assert Catalog.identifier_to_tuple(namespace) in ns_list
-        # Test individual namespace list
-        assert len(one_namespace := catalog.list_namespaces(namespace)) == 1
-        assert Catalog.identifier_to_tuple(namespace) == one_namespace[0]
+    expected_list: list[tuple[str, ...]] = [("db",), ("db2",), ("db%",)]
+    for ns in expected_list:
+        assert ns in ns_list
+
+    ns_list = catalog.list_namespaces("db")
+    expected_list = [("db", "ns1"), ("db", "ns2")]
+    assert len(ns_list) == len(expected_list)
+    for ns in expected_list:
+        assert ns in ns_list
+
+    ns_list = catalog.list_namespaces("db.ns1")
+    expected_list = [("db", "ns1", "ns2")]
+    assert len(ns_list) == len(expected_list)
+    for ns in expected_list:
+        assert ns in ns_list
+
+    ns_list = catalog.list_namespaces("db.ns1.ns2")
+    assert len(ns_list) == 0
 
```

Comment:
heres a good test for the fuzzy match behavior

```suggestion
@pytest.mark.parametrize(
    "catalog",
    [
        lazy_fixture("catalog_memory"),
        lazy_fixture("catalog_sqlite"),
    ],
)
def test_list_namespaces_fuzzy_match(catalog: SqlCatalog) -> None:
    namespace_list = ["db.ns1", "db.ns1.ns2", "db.ns2", "db.ns1X.ns3"]
    for namespace in namespace_list:
        catalog.create_namespace(namespace)

    assert catalog.list_namespaces("db.ns1") == [("db", "ns1", "ns2")]
```

Reply:
@alessandro-nori could you also add this test?

Reply:
sure, thanks for the reminder. I thought I had added it already 🙇 
Now it's done